### PR TITLE
Use `rollup-plugin-postcss` to bundle and optimize the `fides.js` components CSS

### LINKED ISSUE #3288: Bundle CSS into fides.js script, and add styles to DOM from the package
## Current behavior
Currently, rollup bundles the CSS into a separate output in `fides-js/dist/fides.css`.

For usage in `privacy-center/public/fides-js-components-demo.html`, we have to copy the CSS file over into `privacy-center/public/lib/fides.css` so that the page can [import it directly in the HTML](https://github.com/ethyca/fides/pull/3191/files#diff-c7a560cdd3b3acc85bb64a878ff09070dca77bf5858ac85f454bdc31b0324348R12). 

For usage in `privacy-center/public/fides-js-demo.html` file, we dynamically insert the css from `fides-js/dist/fides.css`,  just like we do the js file from `fides-js/dist/fides.js`. This, specifically, is error-prone. 

## Expected behavior
We should bundle the CSS into `fides-js/dist/fides.js` itself so that we won't need to insert the css at the last minute in the bundle on the privacy center.

## Proposed Approach
1. Create a `declaration.d.ts` file with `declare module "*.module.css";` in `fides-js/src`
2. Add the below to `ConsentBanner.tsx`:

```
import styles from "../lib/banner.module.css" assert { type: "css" };

if (typeof window !== 'undefined') {
  // @ts-ignore
  // Create an empty "constructed" stylesheet
    const sheet = new CSSStyleSheet();
  // Apply a rule to the sheet
  sheet.replaceSync(JSON.stringify(styles));
  document.adoptedStyleSheets = [sheet];
}
```

3. Remove the extra css imports from `privacy-center/public/fides-js-components-demo.html` and `privacy-center/pages/api/fides-js.ts`

This currently does not add the styles appropriately, so will need investigation.
 

## Context
_Originally posted by @NevilleS in https://github.com/ethyca/fides/pull/3191#discussion_r1188527874_

### PR DESCRIPTION
**Pull request by NevilleS**
Closes #3288 

### Code Changes

* [X] Replace `rollup-plugin-css` with `rollup-plugin-postcss`
* [X] Remove the `dist/fides.css` output everywhere it was used
* [X] Rename `lib/overlay.module.css` -> `components/fides.css`
* [X] Enable `cssnano` to minimize the CSS bundle output as much as possible
* [X] Rename `fides-consent-*` to `fides-*` for CSS classnames, ids, etc. for some simplicity
* [X] Add some `npm run clean` targets for convenience

### Steps to Confirm

* [X] Run a `turbo build` and then open `clients/privacy-center/public/fides-js-components-demo.html` and confirm the CSS is still included: 
image

### Pre-Merge Checklist

* [X] All CI Pipelines Succeeded
* Documentation:
  * [X] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [X] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [X] Issue Requirements are Met
* [X] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [X]  For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

This was actually really easy!

The most significant impact is on the bundle size: since this adds the CSS to the `fides.js` bundle inline instead of a separate file, we now add another 2 KB to the bundle size, going from 13.69 KB -> 15.66 KB and triggering our 15KB warning:
```
fides-js:build: src/fides.ts → dist/fides.js...
fides-js:build: ️🚨 WARN: fides.js build is getting large! Gzipped size (15.66 KB) exceeded warning size (15 KB)! If you must, update GZIP_SIZE_* constants in clients/fides-js/rollup.config.mjs
fides-js:build: ┌────────────────────────────────┐
fides-js:build: │                                │
fides-js:build: │   Destination: dist/fides.js   │
fides-js:build: │   Bundle Size:  47.09 KB       │
fides-js:build: │   Minified Size:  47.28 KB     │
fides-js:build: │   Gzipped Size:  15.66 KB      │
fides-js:build: │                                │
fides-js:build: └────────────────────────────────┘
```

### CODE REVIEW

**Comment by NevilleS** [comment]
OK, in my testing of this is works well but _doesn't_ fix the styling issue from our NextJS app:
image

The root cause of that appears to be NextJS stripping things out of the <head> tag, surprisingly: https://github.com/vercel/next.js/issues/11012

I think we need to find a way to play nice with NextJS - either natively (unlikely?) or in the Cookie House specifically.

**Comment by NevilleS** [comment]
Created https://github.com/ethyca/fides/issues/3437 to track the rendering issue in Cookie House.

**Comment by NevilleS** [comment]
Used `cssnano` to squeeze another 0.4 KB out of the gzipped size (1.5KB pre-gzip):
```
fides-js:build: ┌────────────────────────────────┐
fides-js:build: │                                │
fides-js:build: │   Destination: dist/fides.js   │
fides-js:build: │   Bundle Size:  45.54 KB       │
fides-js:build: │   Minified Size:  45.4 KB      │
fides-js:build: │   Gzipped Size:  15.27 KB      │
fides-js:build: │                                │
fides-js:build: └────────────────────────────────┘
```

Still above 15K which makes me 😢. I tried a bit of extra optimization like search & replacing the CSS prefix `--fides-consent-overlay` with `--fides-overlay`, which cuts another 0.02 KB gzipped (0.57 KB uncompressed):
```
fides-js:build: ┌────────────────────────────────┐
fides-js:build: │                                │
fides-js:build: │   Destination: dist/fides.js   │
fides-js:build: │   Bundle Size:  44.97 KB       │
fides-js:build: │   Minified Size:  44.84 KB     │
fides-js:build: │   Gzipped Size:  15.25 KB      │
fides-js:build: │                                │
fides-js:build: └────────────────────────────────┘
```

Those kinds of optimizations are pretty ineffective since gzip is much more clever than I am, but I still think we *should* reduce our CSS class names / var names just for comprehensibility. I'll wait for https://github.com/ethyca/fides/pull/3430- and https://github.com/ethyca/fides/pull/3429 to land first because the merge conflicts will be vicious otherwise!

**Comment by NevilleS** [comment]
Waiting for these two PRs to land before finishing this:
1. https://github.com/ethyca/fides/pull/3429
2. https://github.com/ethyca/fides/pull/3430

This is just to let those (more important!) ones land before I rename CSS files and squeeze out a few more bytes of optimizations 🙃

**Review by RobertKeyser — APPROVED** [review]
Looks good to me.

File: clients/fides-js/package.json [hunk 1]
```diff
@@ -47,13 +47,15 @@
...
     "jest-environment-jsdom": "^29.5.0",
     "js-cookie": "^3.0.5",
+    "postcss": "^8.4.24",
     "prettier": "^2.8.8",
     "rollup": "^3.21.3",
     "rollup-plugin-copy": "^3.4.0",
     "rollup-plugin-dts": "^5.3.0",
     "rollup-plugin-esbuild": "^5.0.0",
     "rollup-plugin-filesize": "^10.0.0",
     "rollup-plugin-import-css": "^3.2.1",
```
Comment (eastandwestwind) [inline clients/fides-js/package.json:57]: we can remove this old plugin

**Review by eastandwestwind — CHANGES_REQUESTED** [review]

File: clients/fides-js/package.json (hunk 1 above)
Comment (NevilleS) [reply clients/fides-js/package.json:57]: Oh odd I thought I did that. Thanks

Comment (NevilleS) [reply clients/fides-js/package.json]: Done

**Comment by NevilleS** [comment]
Did a few CSS renames to try to squeeze a few more bytes out of the build, not much impact:
```
fides-js:build: ┌────────────────────────────────┐
fides-js:build: │                                │
fides-js:build: │   Destination: dist/fides.js   │
fides-js:build: │   Bundle Size:  46.05 KB       │
fides-js:build: │   Minified Size:  45.91 KB     │
fides-js:build: │   Gzipped Size:  15.52 KB      │
fides-js:build: │                                │
fides-js:build: └────────────────────────────────┘
```

**Pull request merged**